test/TTBasic/generatorBasic.js

```diff
@@ -0,0 +1,62 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+var log;
+if ( typeof telemetryLog === 'undefined' ) {
+    log =  function(msg, shouldWrite) {
+        if (shouldWrite) {
+            WScript.Echo(msg);
+        }
+    };
+
+}
+else {
+    log = telemetryLog;
+}
+
+var writeTTDLog;
+if (typeof emitTTDLog === 'undefined') {
+    writeTTDLog = function (uri) {
+        // no-op
+    };
+}
+else {
+    writeTTDLog = emitTTDLog;
+}
+/////////////////
+
+
+function* testGenerator() {
+    var i = 0;
+    yield i++;
+    yield i++;
+    yield i++;
+    yield i++;
+}
+
+var gen = testGenerator();
+
+function yieldOne() {
+    var v1 = gen.next();

```

Comment:
Could probably do with some tests that make use of the fact that the `yield` expression in a generator can be given a value by `gen.next(val)`, and make sure that is captured correctly as well. 

can/should we also force some TTD snapshots mid-generator execution?

Reply:
ok, added additional test cases that verify the following:
  - snapshot of a completed generator
  - restore of a completed generator
  - calling writeTTDLog during generator execution
  - restoring at different log points and esnuring expected behavior based on that log point.

Reply:
Just took a look and I don't see any tests for something like

```js
function* gen() {
  var x = yield 1;
  return x;
}

var it = gen();
console.log(it.next());
console.log(it.next(2));
```

which should return values 1 and 2 respectively.

Reply:
OK, added tests for this.  The tests will replay from all generated snap points, and on this branch, the last snap point fails.  This failure is fixed in the branch that contains async-await support, so leaving it commented out in here.